llvm/lib/Target/RISCV/RISCVInstrInfo.cpp

```diff
@@ -897,6 +897,74 @@ MachineInstr *RISCVInstrInfo::foldMemoryOperandImpl(
       .addImm(0);
 }
 
+unsigned getLoadPredicatedOpcode(unsigned Opcode) {
+  switch (Opcode) {
+  case RISCV::LB:
+    return RISCV::PseudoCCLB;
+  case RISCV::LBU:
+    return RISCV::PseudoCCLBU;
+  case RISCV::LH:
+    return RISCV::PseudoCCLH;
+  case RISCV::LHU:
+    return RISCV::PseudoCCLHU;
+  case RISCV::LW:
+    return RISCV::PseudoCCLW;
+  case RISCV::LWU:
+    return RISCV::PseudoCCLWU;
+  case RISCV::LD:
+    return RISCV::PseudoCCLD;
+  default:
+    return 0;
+  }
+}
+
+MachineInstr *RISCVInstrInfo::foldMemoryOperandImpl(
```

Comment:
does it make sense to do the same folding in the FrameIndex variant?

Reply:
> > I hope that can be a follow-up, potentially not immediately?

yeah I'm also fine if it's a follow-up change. In principle, I'm a bit worried about missing the folding opportunities when the spiller (which seems to be the only user of this TII hook) presents the reload value in the form of frame index rather than an already materialized load instruction.